doc/mgmt/SONiC_IntfRangeFeature_HLD.md

```diff
@@ -0,0 +1,235 @@
+mac                Configure MAC parameters   
+mtu                Configure MTU
+nat-zone           NAT Zone   
+neigh_suppress     Enable ARP and ND Suppression      
+no                 Negate a command or set its defaults
+service-policy     Apply ingress or egress policy
+
+sonic(conf-if-range-Vl-1-20)# mtu 9000
+sonic(conf-if-range-Vl-1-20)#
+```
+###### Create Vlan interfaces in given range
+Syntax: `interface range create Vlan Vlan-id - Vlan-id[,Vlan-id - Vlan-id...]`
+```
+sonic(config)# interface range create ?
+Vlan             Vlan interface range
+PortChannel      PortChannel interface range
+
+sonic(config)# interface range create Vlan 3-4,10,11
+sonic(conf-if-range-Vl-3-4,10,11)#
+```
+###### Delete Vlan interfaces existing in given range
+Syntax: `no interface Vlan Vlan-id - Vlan-id[,Vlan-id - Vlan-id...]`
+```
+sonic(config)# no interface Vlan 1-20
+sonic(config)#
+```
+#### 3.6.2.1.3 PortChannel interface range:
+###### Configure PortChannel interfaces existing in given range
+Syntax: `interface range PortChannel Po-id - Po-id[,Po-id - Po-id...]`
+```
+sonic(config)# interface range PortChannel 1-40
+sonic(conf-if-range-Po-1-40)# ?
+end             Exit to the exec Mode
+exit            Exit from current mode
+link            Interface link          
+mac             Configure MAC parameters   
+mclag           Configure MCLAG interface  
+mtu             Configure MTU    
+nat-zone        NAT Zone         
+no              Negate a command or set its defaults
+service-policy  Apply ingress or egress policy     
+shutdown        Disable the interface              
+spanning-tree   Configure spanning tree on port-channel   
+switchport      Configure switchport parameters          
+
+sonic(conf-if-range-Po-1-20)# mtu 9000
+sonic(conf-if-range-Po-1-20)#
+```
+###### Create PortChannel interfaces in given range
+Syntax: `interface range create PortChannel Po-id -Po-id[,Po-id - Po-id...]`
+```
+sonic(config)# interface range create ?
+Vlan             Vlan interface range
+PortChannel      PortChannel interface range
+
+sonic(config)# interface range create PortChannel 3-4,10,11
+sonic(conf-if-range-Po-3-4,10,11)#
+```
+###### Delete PortChannel interfaces existing in given range
+Syntax: `no interface range PortChannel Po-id - Po-id[,Po-id - Po-id...]`
+```
+sonic(config)# no interface Portchannel 1-30
+sonic(config)#
+```
+####  3.6.2.1.4 Show interfaces existing in given range
+```
+sonic# show interface Ethernet 0-20
+sonic# show interface status Ethernet 0-20,40
+sonic# show interface counters Ethernet 0-20
+```
+#### 3.6.2.2 Interface range Commands in alias mode
+##### Configuring a Range of Interfaces
+```
+sonic(config)# interface range e1/2-e1/10
+sonic(conf-if-range-Eth1/2-1/10)# mtu 9000
+```
+#### 3.6.2.3 Debug Commands
+#### 3.6.2.4 IS-CLI Compliance
+### 3.6.3 REST API Support
+Not supported
+#### 3.6.3.1 GET
+
+# 4 Flow Diagrams
+
+# 5 Error Handling
+
+# 6 Serviceability and Debug
+# 7 Warm Boot Support
```

Comment:
Need to document the upgrade/downgrade behavior? Will there be any impact if buzznikplus build with range config is downgraded to buzznikMR and viceversa

Reply:
What impact does upgrade/downgrade have with range support ?

Reply:
How are the range command showup in running config?
If we downgrade to a build doesnt support range (with saved config having range command), what will happen.
So such behavour needs to be documented. If there is no impact, then lets add that that there wont be any impact

Reply:
show running config just shows the individual interfaces config, interface can be configured with/without using range command.  So config done using range cmd should not have any effect on show running config.